File: lcy8047/14_LongestCommonPrefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        std::string common_prefix = strs[0];
        for ( auto s : strs )
        {
            for ( size_t i = 0; i < common_prefix.size(); ++ i )
            {
                if ( s[i] != common_prefix[i] )
                {
                    size_t common_prefix_size = common_prefix.size();
                    for ( size_t k = 0; k < common_prefix_size - i; ++ k )
                    {
                        common_prefix.pop_back();
                    }
                    break;
                }
            }
            if ( common_prefix.size() == 0 )
            {
                return "";
            }
        }
        return common_prefix;
    }
};

File: EunjiShin/LeetCode/14.longest-common-prefix.java
package Easy;

import java.util.Arrays;
import java.util.Comparator;

/*
 * @lc app=leetcode id=14 lang=java
 *
 * [14] Longest Common Prefix
 */

// @lc code=start
class Solution {

    public String longestCommonPrefix(String[] strs) {
        if (strs.length == 1) {
            return strs[0];
        }

        String result = strs[0];
        for (int i = 1; i < strs.length; i++) {
            while (strs[i].indexOf(result) != 0) {
                result = result.substring(0, result.length() - 1);
            }
        }
        return result;
    }

}
// @lc code=end
